### PR TITLE
Change the debuglet api to make it easier to test

### PR DESCRIPTION
**Pull request by ofrobots**
@matthewloring ptal.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -33,7 +33,8 @@ if (process.env.hasOwnProperty('GCLOUD_DEBUG_DISABLE')) {
 }
 
 if (config.enabled) {
-  agent.start(config, logger.create(config.logLevel, '@google/cloud-debug'),
-      module.exports);
+  var debuglet = new Debuglet(
+    config, logger.create(config.logLevel, '@google/cloud-debug'));
+  debuglet.start();
```
Comment (matthewloring) [inline index.js]: Is there a reason we are not interested in errors coming from this call?

**Comment by matthewloring** [comment]
LGTM but I agree that the callback to start should be eliminated when possible.

**Comment by ofrobots** [comment]
There have been enough changes to warrant another look.

File: lib/debuglet.js [hunk 2]
```diff
@@ -31,22 +34,14 @@ var NODE_VERSION_MESSAGE = 'V8 Debug API not available – make sure you are ' +
...
-    exports.private_ = agent;
-  }
-};
+module.exports = Debuglet;
 
 /**
  * @param {Object} config The option parameters for the Debuglet.
+ * @event 'error'
+ * @event 'started'
+ * @event 'registered'
```
Comment (matthewloring) [inline lib/debuglet.js]: Does registered have the connotation of "successfully initialized"? Could we document what an event of this type means.

Comment (matthewloring) [inline lib/debuglet.js]: Ah, it looks like `started` is the initialized event. Could we document what registered means.

**Comment by matthewloring** [comment]
LGTM with small nit. I like the event emitter structure a lot more!

File: lib/debuglet.js (hunk 2 above)
Comment (ofrobots) [inline lib/debuglet.js]: done.

**Pull request merged**